vdyp-lib/vdyp-common/src/main/java/ca/bc/gov/nrs/vdyp/model/BaseVdypSpecies.java

```diff
@@ -12,31 +13,40 @@
 
 public abstract class BaseVdypSpecies<I extends BaseVdypSite> {
 	private final PolygonIdentifier polygonIdentifier; // FIP_P/POLYDESC
-	private final LayerType layerType; // This is also represents the distinction between data stored in
+
+	// This is also represents the distinction between data stored in
 	// FIPL_1(A) and FIP_V(A). Where VDYP7 stores both and looks at certain values
 	// to determine if a layer is "present". VDYP8 stores them in a map keyed by
-	// this value
+	// this value.
+	private final LayerType layerType;
 
 	private final String genus; // FIPSA/SP0V
 
+	/** the species index within species definition file (e.g. SPODEF_v0.dat) */
+	private final int genusIndex; // BANK1/ISPB, L1COM1/ISPL1, etc.
+
 	private float percentGenus; // FIPS/PCTVOLV L1COM1/PCTL1
 
 	// This is computed from percentGenus, but VDYP7 computes it in a way that might
 	// lead to a slight difference so it's stored separately and can be modified.
 	@Computed
 	private float fractionGenus; // FRBASP0/FR
 
-	private Map<String, Float> speciesPercent; // Map from
+	private Sp64DistributionSet sp64DistributionSet;
+
 	private Optional<I> site;
 
 	protected BaseVdypSpecies(
-			PolygonIdentifier polygonIdentifier, LayerType layerType, String genus, float percentGenus, Optional<I> site
+			PolygonIdentifier polygonIdentifier, LayerType layerType, String genus, int genusIndex, float percentGenus,
+			Sp64DistributionSet sp64DistributionSet, Optional<I> site
 	) {
 		this.polygonIdentifier = polygonIdentifier;
 		this.layerType = layerType;
 		this.genus = genus;
 		this.setPercentGenus(percentGenus);
 		this.site = site;
+		this.genusIndex = genusIndex;
```

Comment:
I dislike the idea of having both the string ID and the numeric index and being able to set them separately so they might end up mismatched.

Maybe if we do need both, give the builder a copy of the control map and on build it fills in whichever of the two hasn't been provided  If both are provided either check they match, or use whichever was set most recently (Both setters set the other value to empty)

Reply:
I didn't like it either - glad you prompted me to think of a better way.